src/topology/algebra/continuous_monoid_hom.lean

```diff
@@ -231,16 +233,93 @@ end⟩⟩
+(is_inducing A C).continuous_iff.2 $ f.to_continuous_map.continuous_comp_left.comp
+  (is_inducing B C).continuous
+
+@[to_additive] lemma continuous_comp_right (f : continuous_monoid_hom B C) :
+  continuous (λ g : continuous_monoid_hom A B, f.comp g) :=
+(is_inducing A C).continuous_iff.2 $ f.to_continuous_map.continuous_comp.comp
+  (is_inducing A B).continuous
+
+variables (E)
+
+/-- `continuous_monoid_hom _ f` is a functor. -/
+@[to_additive "`continuous_add_monoid_hom _ f` is a functor."]
+def comp_left (f : continuous_monoid_hom A B) :
+  continuous_monoid_hom (continuous_monoid_hom B E) (continuous_monoid_hom A E) :=
+{ to_fun := λ g, g.comp f,
+  map_one' := rfl,
+  map_mul' := λ g h, rfl,
+  continuous_to_fun := f.continuous_comp_left }
+
+variables (A) {E}
+
+/-- `continuous_monoid_hom f _` is a functor. -/
+@[to_additive "`continuous_add_monoid_hom f _` is a functor."]
+def comp_right {B : Type*} [comm_group B] [topological_space B]
+  [topological_group B] (f : continuous_monoid_hom B E) :
+  continuous_monoid_hom (continuous_monoid_hom A B) (continuous_monoid_hom A E) :=
+{ to_fun := λ g, f.comp g,
+  map_one' := ext (λ a, map_one f),
+  map_mul' := λ g h, ext (λ a, map_mul f (g a) (h a)),
+  continuous_to_fun := f.continuous_comp_right }
+
 end continuous_monoid_hom
 
 /-- The Pontryagin dual of `A` is the group of continuous homomorphism `A → circle`. -/
 @[derive [topological_space, t2_space, comm_group, topological_group, inhabited]]
 def pontryagin_dual := continuous_monoid_hom A circle
+variables {A B C D E}
+
+noncomputable instance : continuous_monoid_hom_class (pontryagin_dual A) A circle :=
+continuous_monoid_hom.continuous_monoid_hom_class
+
+namespace continuous_monoid_hom
+
+/-- `pontryagin_dual` is a functor. -/
+protected noncomputable def dual (f : continuous_monoid_hom A B) :
+  continuous_monoid_hom (pontryagin_dual B) (pontryagin_dual A) :=
+f.comp_left circle
+
+@[simp] lemma dual_apply (f : continuous_monoid_hom A B) (x : pontryagin_dual B) (y : A) :
+  f.dual x y = x (f y) :=
+rfl
+
+@[simp] lemma dual_one : (one A B).dual = one (pontryagin_dual B) (pontryagin_dual A) :=
+ext (λ x, ext (λ y, map_one x))
+
+@[simp] lemma dual_comp (g : continuous_monoid_hom B C) (f : continuous_monoid_hom A B) :
+  (comp g f).dual = comp f.dual g.dual :=
+ext (λ x, ext (λ y, rfl))
+
+@[simp] lemma dual_mul (f g : continuous_monoid_hom A E) : (f * g).dual = f.dual * g.dual :=
+ext (λ x, ext (λ y, map_mul x (f y) (g y)))
+
+variables (A B C D E)
+
+/-- `continuous_monoid_hom.dual` as a `continuous_monoid_hom`. -/
+noncomputable def dual_hom [locally_compact_space E] :
+  continuous_monoid_hom (continuous_monoid_hom A E)
+    (continuous_monoid_hom (pontryagin_dual E) (pontryagin_dual A)) :=
+{ to_fun := continuous_monoid_hom.dual,
+  map_one' := dual_one,
+  map_mul' := dual_mul,
+  continuous_to_fun := (is_inducing _ _).continuous_iff.mpr
+    (continuous_map.continuous_of_continuous_uncurry _ continuous_comp) }
```

Comment:
This suggests that we may want a variant of `continuous_of_continuous_uncurry` for `continuous_monoid_hom`